src/Haskell/Ide/Engine/Options.hs

```diff
@@ -6,11 +6,13 @@ import           Options.Applicative.Simple
 data GlobalOpts = GlobalOpts
   { optDebugOn       :: Bool
   , optLogFile       :: Maybe String
-  , optLsp           :: Bool -- Kept for a while, to not break legacy clients
+  , optLsp           :: Bool
   , projectRoot      :: Maybe String
   , optBiosVerbose   :: Bool
   , optCaptureFile   :: Maybe FilePath
   , optExamplePlugin :: Bool
+  , optDryRun        :: Bool
+  , optFiles         :: [FilePath]
```

Comment:
better name required.

Reply:
`optSanityCheck`? `optProjectLoadCheck`?

Can you list directories as well, in `optFiles`, and have the recursive contents checked?

Reply:
Yeah. If it is a directory, it is recursed into it and if it is a FilePath, its existence is checked and filtered if it does not exist. Which is maybe not the desired behaviour.

Reply:
Following question remain for me:
Should `--dry-run` be the default? So that it doesnt by default try to load every file in the project?

Reply:
I don't think it's that dangerous to load every file in the project, it's not like its destructively editing anything. I'm happy if it loads them by default